code/modules/sql/database_player_age.dm

```diff
@@ -57,11 +46,9 @@
 	var/sql_computerid = sql_sanitize_text(src.computer_id)
 	var/sql_admin_rank = sql_sanitize_text(admin_rank)
 
-	if(player_exists)
-		//Player already identified previously, we need to just update the 'lastseen', 'ip' and 'computer_id' variables
-		var/database/query/query_update = new("UPDATE player SET lastseen = julianday('now'), ip = '[sql_ip]', computerid = '[sql_computerid]', lastadminrank = '[sql_admin_rank]' WHERE ckey = [sql_ckey]")
-		query_update.Execute(global_db)
-	else
-		//New player!! Need to insert all the stuff
-		var/database/query/query_insert = new("INSERT INTO player VALUES ('[sql_ckey]', julianday('now'), julianday('now'), '[sql_ip]', '[sql_computerid]', '[sql_admin_rank]')")
-		query_insert.Execute(global_db)
\ No newline at end of file
+	var/database/query/query_update
+	if(player_exists) //Player already identified previously, we need to just update the 'lastseen', 'ip' and 'computer_id' variables
+		query_update = new("UPDATE player SET lastseen = datetime('now'), ip = '?', computerid = '?', lastadminrank = '?' WHERE ckey = '?';", sql_ip, sql_computerid, sql_admin_rank, sql_ckey)
```

Comment:
This is double-sanitised now, using prepared statements sanitises the data internally, just feed it the raw data.

```dm
/* REMOVE THIS CODE
	var/sql_ckey =       sql_sanitize_text(ckey(key))
	var/sql_ip =         sql_sanitize_text(src.address)
	var/sql_computerid = sql_sanitize_text(src.computer_id)
	var/sql_admin_rank = sql_sanitize_text(admin_rank)
*/

	var/database/query/query_update
	if(player_exists) //Player already identified previously, we need to just update the 'lastseen', 'ip' and 'computer_id' variables
		query_update = new("UPDATE player SET lastseen = datetime('now'), ip = '?', computerid = '?', lastadminrank = '?' WHERE ckey = '?';", address, computer_id, admin_rank, ckey) // params changed
```